File: backend/app/schemas/curriculum.py
from pydantic import BaseModel, Field
from typing import List, Optional

class Topic(BaseModel):
    topic_id: str
    name: str
    description: str
    learning_objectives: List[str] = Field(default_factory=list)
    key_tools: List[str] = Field(default_factory=list)

class CurriculumDay(BaseModel):
    day_number: int
    day_id: str
    title: str
    topics: List[Topic] = Field(default_factory=list)

class Module(BaseModel):
    module_id: str
    title: str
    days: List[CurriculumDay] = Field(default_factory=list)

class Curriculum(BaseModel):
    curriculum_id: str
    title: str
    description: str
    version: str = "1.0.0"
    modules: List[Module] = Field(default_factory=list)

File: frontend/src/types/curriculum.ts
export interface Topic {
  topic_id: string;
  name: string;
  description: string;
  learning_objectives: string[];
  key_tools: string[];
}

export interface CurriculumDay {
  day_number: number;
  day_id: string;
  title: string;
  topics: Topic[];
}

export interface Module {
  module_id: string;
  title: string;
  days: CurriculumDay[];
}

export interface Curriculum {
  curriculum_id: string;
  title: string;
  description: string;
  version: string;
  modules: Module[];
}
